### PR TITLE
ETC Gotham Fork

### PR DESCRIPTION
**Pull request by edwardmack**
## PR description
This PR introduces support for the Gotham fork on the Ethereum Classic Network including:
- ECIP-1017: Monetary Policy and Final Modification to the Ethereum Classic Emission Schedule
- ECIP 1039: Monetary policy rounding specification
- ECIP 1041: Remove Difficulty Bomb
## Fixed Issue(s)

N/a

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+      final boolean skipZeroBlockRewards) {
+    this.transactionProcessor = transactionProcessor;
+    this.transactionReceiptFactory = transactionReceiptFactory;
+    this.blockReward = blockReward;
+    this.miningBeneficiaryCalculator = miningBeneficiaryCalculator;
+    this.skipZeroBlockRewards = skipZeroBlockRewards;
+  }
+
+  @Override
+  public Result processBlock(
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java]: This should be refactored into an AbstractBlockProcessor.  Between this one and MainnetBlockProcessor this method is identical, aside from fully qualified references to `Result`.  That class should be in the parent too.  `rewardCoinbase` should be an abstract method and Mainnet and Classic should do their calculation in the subclasses.

Comment (edwardmack) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicBlockProcessor.java]: I agree.... This has been updated.

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    return AbstractBlockProcessor.Result.successful(receipts);
+  }
+
+  abstract boolean rewardCoinbase(
+      final MutableWorldState worldState,
+      final ProcessableBlockHeader header,
+      final List<BlockHeader> ommers,
+      final boolean skipZeroBlockRewards);
+
+  //    private boolean rewardCoinbase(
```
Comment (shemnon) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/AbstractBlockProcessor.java]: This commented segment snuck in.

**Comment by shemnon** [comment]
CircleCI failures feel like flakey tests to me.  The commit to remove the commented out code in the master should kick the build again.

**Pull request merged**